core/node/node_framework/src/implementations/layers/external_price_api_client.rs

```diff
@@ -0,0 +1,44 @@
+use std::sync::Arc;
+
+use zksync_config::configs::ExternalPriceApiClientConfig;
+use zksync_external_price_api::coingecko_api::CoinGeckoPriceAPIClient;
+
+use crate::{
+    implementations::resources::price_api_client::PriceAPIClientResource,
+    service::ServiceContext,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+/// Wiring layer for `ExternalPriceApiClient`
+///
+/// Responsible for inserting all the resources to get base token prices from external price feeds to
+/// be used by the `BaseTokenRatioPersister`.
+///
+/// ## Adds resources
+///
+/// - PriceAPIClientResource
+#[derive(Debug)]
+pub struct ExternalPriceApiClientsLayer {
```

Comment:
An appropriate name would be `CoingeckoClientLayer` (e.g. we're adding a concrete layer here, not abstract).

Reply:
I was thinking about having
- one resource per external API price provider the chain operator wants to use
- one layer to insert them all
Do you think that's not the way to go? Would it be more appropriate for there to be a layer per external price provider?

Reply:
Layer per external price provider is more flexible. If you have one layer per concrete resource implementation, then you need to choose _just_ layer. If you have layer with several resources changed via "if", then you have to pass the config to layer so that it can choose the resource for you. If we look further, someone may implement their own external API source as a separate crate (one of the problems we're trying to solve with the framework is that we don't want to merge every single feature to our repo, we want to enable people to write their own layers and inject them into our code), you'll get in a weird situation where a few resource implementations are added by layer A, but for others you need B, C, D.

tl;dr One concrete resource per layer is a good option. [API example](https://github.com/matter-labs/zksync-era/tree/main/core/node/node_framework/src/implementations/layers/web3_api/tx_sink), [consensus example](https://github.com/matter-labs/zksync-era/tree/main/core/node/node_framework/src/implementations/layers/consensus).